src/sysfiles.c

```diff
@@ -800,7 +816,7 @@ Int SyFopen (
     else if ( strcmp( name, "*errin*" ) == 0 ) {
         if ( strcmp( mode, "r" ) != 0 )
           return -1;
-        else if ( syBuf[2].fp == -1 )
+        else if ( !SyBufInUse( 2 ) )
```

Comment:
off-topic, but one other thing we may want to do (I thought about this before) is to add an enum for the special values 0, 1, 2, 3 (stdin/out, errin/out). Say `GAP_STDIN` or `ID_STDIN` etc. ?